app/controllers/users/confirmations_controller.rb

```diff
@@ -112,7 +112,12 @@ def after_confirmation_path_for(resource)
     end
 
     def process_confirmed_user
-      analytics.track_event('Email changed and confirmed', @confirmable)
+      if @confirmable.previous_changes.key?('email')
+        analytics.track_event('Email change confirmed', @confirmable)
+        create_user_event(:email_changed, @confirmable)
+      else
+        analytics.track_event('Email address confirmed', @confirmable)
```

Comment:
This is not necessary either. This method only processes one scenario: when the user is already confirmed, meaning they already have an email address, and then they change it and then confirm it.

The scenario where a user confirms their email for the first time is handled by the `process_unconfirmed_user` method.